File: LeetCode/ProblemSet/001.Two Sum/two-sum.js
"use strict";

/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
    let map = new Map();
    for (let i = 0, len = nums.length; i < len; i++) {
        let a = target - nums[i];
        if (map.has(a)) {
            return [map.get(a), i];
        }
        map.set(nums[i], i);
    }
    return null;
};

console.log(twoSum([2, 7, 11, 15], 9));
console.log(twoSum([3, 3], 6));

File: LeetCode/Go-Solutions/ProblemSet/0001.Two_Sum/two_sum.go
package two_sum

func twoSum(nums []int, target int) []int {
	visited := make(map[int]int)
	for k, v := range nums {
		if i, ok := visited[target-v]; ok {
			return []int{i, k}
		} else {
			visited[v] = k
		}
	}
	return nil
}
